ot/dr.py

```diff
@@ -376,3 +381,153 @@ def Vpi(X, Y, a, b, pi):
         iter = iter + 1
 
     return pi, U
+
+
+def ewca(X, U0=None, reg=1, k=2, method='BCD', sinkhorn_method='sinkhorn', stopThr=1e-6, maxiter=100, maxiter_sink=1000, maxiter_MM=10, verbose=0):
+    r"""
+    Entropic Wasserstein Component Analysis :ref:`[52] <references-entropic-wasserstein-component_analysis>`.
+
+    The function solves the following optimization problem:
+
+    .. math::
+        \mathbf{U} = \mathop{\arg \min}_\mathbf{U} \quad
+        W(\mathbf{X}, \mathbf{U}\mathbf{U}^T \mathbf{X})
+
+    where :
+
+    - :math:`\mathbf{U}` is a matrix in the Stiefel(`p`, `d`) manifold
+    - :math:`W` is entropic regularized Wasserstein distances
+    - :math:`\mathbf{X}` are samples
+
+    Parameters
+    ----------
+    X : ndarray, shape (n, d)
+        Samples from measure :math:`\mu`.
+    U0 : ndarray, shape (d, k), optional
+        Initial starting point for projection.
+    reg : float, optional
+        Regularization term >0 (entropic regularization).
+    k : int, optional
+        Subspace dimension.
+    method : str, optional
+        Eather 'BCD' or 'MM' (Block Coordinate Descent or Majorization-Minimization).
+        Prefer MM when d is large.
+    sinkhorn_method : str
+        Method used for the Sinkhorn solver, see :ref:`ot.bregman.sinkhorn` for more details.
+    stopThr : float, optional
+        Stop threshold on error (>0).
+    maxiter : int, optional
+        Maximum number of iterations of the BCD/MM.
+    maxiter_sink : int, optional
+        Maximum number of iterations of the Sinkhorn solver.
+    maxiter_MM : int, optional
+        Maximum number of iterations of the MM (only used when method='MM').
+    verbose : int, optional
+        Print information along iterations.
+
+    Returns
+    -------
+    pi : ndarray, shape (n, n)
+        Optimal transportation matrix for the given parameters.
+    U : ndarray, shape (d, k)
+        Matrix Stiefel manifold.
+
+
+    .. _references-entropic-wasserstein-component_analysis:
+    References
+    ----------
+    .. [52] Collas, A., Vayer, T., Flamary, F., & Breloy, A. (2023).
+            Entropic Wasserstein Component Analysis.
+    """  # noqa
+    n, d = X.shape
+    X = X - X.mean(0)
+
+    if U0 is None:
+        pca_fitted = PCA(n_components=k).fit(X)
+        U = pca_fitted.components_.T
+        if method == 'MM':
+            lambda_scm = pca_fitted.explained_variance_[0]
+    else:
+        U = U0
+
+    # marginals
+    u0 = (1. / n) * np.ones(n)
+
+    # print iterations
+    if verbose > 0:
+        print('{:4s}|{:13s}|{:12s}|{:12s}'.format('It.', 'Loss', 'Crit.', 'Thres.') + '\n' + '-' * 40)
+
+    def compute_loss(M, pi, reg):
+        return np.sum(M * pi) + reg * np.sum(pi * (np.log(pi) - 1))
+
+    def Grassmann_distance(U1, U2):
```

Comment:
don't make exceptions to snake_case convention
```suggestion
    def grassmann_distance(U1, U2):
```